syntax/std_seq.go

```diff
@@ -64,5 +65,140 @@ func stdSeq() rel.Attr {
 	return rel.NewTupleAttr("seq",
 		rel.NewNativeFunctionAttr("concat", stdSeqConcat),
 		rel.NewNativeFunctionAttr("repeat", stdSeqRepeat),
+		createNestedFuncAttr("contains", 2, func(args ...rel.Value) rel.Value {
+			return process("contains", args...)
+		}),
+		createNestedFuncAttr("split", 2, func(args ...rel.Value) rel.Value {
+			return process("split", args...)
+		}),
+		createNestedFuncAttr("sub", 3, func(args ...rel.Value) rel.Value {
+			return process("sub", args...)
+		}),
+		createNestedFuncAttr("has_prefix", 2, func(args ...rel.Value) rel.Value {
+			return process("has_prefix", args...)
+		}),
+		createNestedFuncAttr("has_suffix", 2, func(args ...rel.Value) rel.Value {
+			return process("has_suffix", args...)
+		}),
+		createNestedFuncAttr("join", 2, func(args ...rel.Value) rel.Value {
+			return process("join", args...)
+		}),
 	)
 }
+
+func process(apiName string, args ...rel.Value) rel.Value {
+	// As https://github.com/arr-ai/arrai/issues/230, the subject parameter is placed in the last param in API signature,
+	// so get handler by the last param.
+	handler := handlerMapping[typeMethod{reflect.TypeOf(args[len(args)-1]), apiName}]
+	return handler(args...)
+}
+
+// This is seq API handlders mapping.
+var (
+	handlerMapping = map[typeMethod]func(...rel.Value) rel.Value{
+		// API contains
+		{reflect.TypeOf(rel.String{}), "contains"}: func(args ...rel.Value) rel.Value {
+			return rel.NewBool(strings.Contains(mustAsString(args[1]), mustAsString(args[0])))
+		},
```

Comment:
any reason to use `reflect`? A `switch` type can do the same thing

```go
switch value.(type){
case rel.String:
   ...
case rel.Array:
   ...
case rel.Bytes:
   ...
}

Reply:
I tried to avoid switch code, so added reflect to check args type and find the correct handler function. Which can make `rel.NewTupleAttr("seq", ...)` code simple, it will be looks like:
```
createNestedFuncAttr("has_prefix", 2, func(args ...rel.Value) rel.Value {
	return process(args)
}),
...
```
And there was a mapping between functions and `args...` types with `reflect`. So if there were some new types to be added, it could update the mapper and don't need to update `rel.NewTupleAttr("seq", ...)`. It could say this mapper was a code extension point.

And finally, I found it was not good as:
1. Code with statement `switch` is more easy to read and understand the rule.
2. The mapper was tedious as it had a lot of function signature which added the total code lines in this file.
3. The extension point is not really necessary now.

So I have changed and used `switch` statement.